### ISSUE TITLE
Themes: links to /design do not work

### ISSUE THREAD

**Issue by designsimply** [Automattic/wp-calypso#20416]
Steps to reproduce:

1. Go to https://wordpress.com/design

<strong>Result:</strong> I expected to see Themes or be prompted to select a site where I can view themes but an empty page loads instead.

<strong>Note:</strong> in the left sidebar, Themes points to `/themes` but used to point to `/design`. Can we add a redirect to help keep old links working?

![screen shot 2017-12-01 at fri dec 1 10 03 09 am](https://user-images.githubusercontent.com/1119271/33493921-dc8a182e-d67e-11e7-9f8d-8fb8d3de6d93.png)
<sup>Seen at https://wordpress.com/design using Firefox 57.0.1 on Mac OS X 10.13.1.</sup>

#### Context / Source

/hat tip @ryanboren for reporting via Slack. (internal ref: p1512143595000075-slack-flow) #reallife  #dogfooding #flowsharing

**Comment by designsimply** [comment]
Also see 7971-wpcom (internal ref).

**Comment by lancewillett** [comment]
Sorry I missed this when it was originally reported. Fixed in r166509-wpcom.

**Comment by lancewillett** [comment]
Re-opening:

The route should be fixed so old URLs still work.

**Comment by designsimply** [comment]
Also see https://github.com/Automattic/wp-calypso/issues/20423

**Comment by lancewillett** [comment]
Investigating with @samouri to see if this is related to https://github.com/Automattic/wp-calypso/pull/20267 user bootstrapping changes. About 8 days ago.

**Comment by lancewillett** [comment]
@seear @ockham Sorry to bug you two... but would you mind helping with a quick investigation? Just trying to find out if this is a recent side effect of other changes (SSR, user-bootstrapping) — or if it's a simple bug with routing or config or something like that.

**Comment by seear** [comment]
Server debug when hitting `/design`:
```
  calypso:server-render cache access for key +32m Cannot read property 'redirect' of undefined
  calypso:server-render Server render time (ms) +1ms 1
GET /design 404 435.735 ms - 33753
```

That error must be from [this line](https://github.com/Automattic/wp-calypso/blob/master/client/my-sites/themes/index.node.js#L23):
```
		router( [ '/design', '/design/*' ], ( { originalUrl, res } ) => {
			res.redirect( 301, '/themes' + originalUrl.slice( '/design'.length ) );
		} );
```

Introduced by the [removal of the `res` field](https://github.com/Automattic/wp-calypso/pull/19319/files#diff-336276ec8a6d8fa6e6ecd99de5a7bab3R70) in #19319.

**Comment by seear** [comment]
Fixed in #20631
